### ISSUE TITLE
[FR/Bug] - "Best image timeout" for clips (0.8.0)

### ISSUE THREAD

**Issue by tubalainen** [blakeblackshear/frigate#346]
I've got quite a lot of short video clips when there is more than one person moving within a camera - like when my kids are playing in the front yard.

Would it be possible to have a "re-trigger" threshold to just continue recording on "the first best detection" rather than creating many short videos? The event/scene is quite hard to follow with so many clips.
Something similar to "best image timeout" for snaps?

I am not 100% sure that I understand the global `max_seconds: 300` setting fully, is it suppose do prevent "re-triggers"?

![image](https://user-images.githubusercontent.com/26795429/101605215-35175800-3a02-11eb-8b9f-c082c00408ea.png)

**Comment by blakeblackshear** [comment]
There are several things going on here. I see something similar on my cameras, but my clips aren't as short. What is the frame rate of your cameras?

**Comment by blakeblackshear** [comment]
max_seconds is a cache setting. It is the maximum clip length to prevent the cache from filling forever during long events.

**Comment by tubalainen** [comment]
> There are several things going on here. I see something similar on my cameras, but my clips aren't as short. What is the frame rate of your cameras?

All cams are 8 fps.
I use a ram based spool area (tmpfs) and the clips folder are pointed towards an USB3.0 external disk.
I see no CPU bottleneck even if I am stressing the machine.

What would be the symptom if there are missing frames or other artifacts in the stream while the clip is being created? Might this be the cause of my short clips? 

AND

I have also noticed that some clips claiming to be "5 s" are much longer, but the clips are not complete/broken (in my case). (See screenshot below).
The (broken?) clip can be downloaded here for your reference, no errors could be found in the frigate logs during this event.
![image](https://user-images.githubusercontent.com/26795429/101643883-de774180-3a34-11eb-96da-f55c6f601cf7.png)

Clip download: http://tuben.lugn.nu:3000/0ad1648910ed

**Comment by blakeblackshear** [comment]
It's not a resource issue. Frigate determines that an object is gone after a certain number of frames go by without it being detected. With higher frame rates, this can cause objects to be marked as gone in less than a second. This would cause some events to be very short. Also, the duration shown in the media browser is the length of the event. It does not include the time for "pre capture" time.

**Comment by rjgrandy** [comment]
I also see this issue, where many short clips of the same "event" show up in the media browser, would like a way to have them combined if possible

**Comment by lawrence-jeff** [comment]
Also fighting this -I would like it to start recording and continue as one recording until a user defined period of no detection occurs - Say 3 minutes, if it clipped the end of the video to correspond to the beginning of that period it would be ideal. I also have kids go in and out of frame and it piles up the videos
